modules/userdirectory/src/main/java/org/opencastproject/userdirectory/InMemoryUserAndRoleProvider.java

```diff
@@ -124,75 +118,90 @@ protected void activate(ComponentContext cc) {
               + "#                                                    #\n"
               + "######################################################");
     }
-
-    // Create the digest user
-    createSystemUsers();
   }
 
   @Override
   public String getName() {
     return PROVIDER_NAME;
   }
 
+  /**
+   * Get a list of all users for the current organization, loading the users from configuration if the organization had
+   * not been loaded before.
+   *
+   * @return List of users.
+   */
+  private List<User> getOrganizationUsers() {
+    Organization organization = securityService.getOrganization();
+    if (!inMemoryUsers.containsKey(organization)) {
```

Comment:
One last thing: 

```
users = inMemoryUsers.get(organization);
if (users == null) {
   createtystemUsers(organization);
   users = inMemoryUsers.get(organization);
}
return users;
```
... would be more efficient for the common case as we don't need to lookup the users in the map twice.

Not a big deal as this data structure won't contain too many users, but anyway, it does save CPU cycles.

Reply:
done